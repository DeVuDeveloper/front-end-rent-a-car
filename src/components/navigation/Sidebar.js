/* eslint-disable arrow-body-style */
/* eslint-disable no-confusing-arrow */
/* eslint-disable react/prop-types */
import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import logo from '../../assets/images/logo.png';
import twitter from '../../assets/images/twitter-icon.png';
import facebook from '../../assets/images/facebook-icon.png';
import linkedin from '../../assets/images/linkedin-icon.png';
import github from '../../assets/images/github-icon.png';
import Logout from '../auth/Logout';
import './sidebar.css';

const social = [
  { icon: twitter },
  { icon: facebook },
  { icon: linkedin },
  { icon: github },
];

const Sidebar = ({ currentUser }) => {
  return (
    <div className="flex flex-col w-64 sidebar-wrapper">
      <span className="nav-header">
        <h2>Rent a Car</h2>
      </span>

      <span className="user-mail">{currentUser.email}</span>

      <div className="flex-1 flex flex-col pt-3 pb-4">
        <div className="flex items-center flex-shrink-0 px-4">
          <a href="/">
            <img src={logo} className="w-40" alt="logo" />
          </a>
        </div>
        <nav className="mt-12 flex-1 desktop-nav" aria-label="Sidebar">
          <div className="pl-3 uppercase font-black text-md">
            <div className="nav-column">
              <NavLink
                to="/home"
                className={({ isActive }) => (isActive ? 'active' : 'inactive')}
              >
                <p>CARS</p>
              </NavLink>
              <NavLink
                to="/reserve"
                className={({ isActive }) => (isActive ? 'active' : 'inactive')}
              >
                <p>RESERVE</p>
              </NavLink>
              <NavLink
                to="/reservations"
                className={({ isActive }) => (isActive ? 'active' : 'inactive')}
              >
                <p>MY RESERVATIONS</p>
              </NavLink>
              {currentUser.role === 'admin' && (
                <NavLink
                  to="/add_car"
                  className={({ isActive }) => isActive ? 'active' : 'inactive'}
                >
                  <p>ADD CAR</p>
                </NavLink>
              )}

              {currentUser.role === 'admin' && (
                <NavLink
                  to="/delete"
                  className={({ isActive }) => isActive ? 'active' : 'inactive'}
                >
                  <p>DELETE CAR</p>
                </NavLink>
              )}

              <span className="logout">
                <Logout />
              </span>
            </div>
          </div>
        </nav>
      </div>
      <div className="flex-shrink-0 flex-200 p-4">
        <div className="flex gap-2">
          {social.map((item) => (
            <a href="/home" key={item.icon}>
              <img src={item.icon} alt="social-icon" className="w-6" />
            </a>
          ))}
        </div>
        <p className="text-gray-700 font-bold mt-2">Â© 2022</p>
      </div>
    </div>
  );
};

const mapStateToProps = ({ auth: { currentUser } }) => {
  return { currentUser };
};

export default connect(mapStateToProps)(Sidebar);
