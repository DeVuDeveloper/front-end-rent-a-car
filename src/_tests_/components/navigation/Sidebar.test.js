import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Sidebar from '../../../components/navigation/Sidebar';
import store from '../../../redux/configureStore';

it('should render Sidebar correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Sidebar />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
