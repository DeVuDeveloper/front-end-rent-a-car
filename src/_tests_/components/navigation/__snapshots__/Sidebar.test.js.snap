// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar correctly 1`] = `
<div
  className="flex flex-col w-64 sidebar-wrapper"
>
  <span
    className="nav-header"
  >
    <h2>
      Rent a Car
    </h2>
  </span>
  <span
    className="user-mail"
  />
  <div
    className="flex-1 flex flex-col pt-3 pb-4 overflow-y-auto"
  >
    <div
      className="flex items-center flex-shrink-0 px-4"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          className="w-40"
          src="logo.png"
        />
      </a>
    </div>
    <nav
      aria-label="Sidebar"
      className="mt-12 flex-1 desktop-nav"
    >
      <div
        className="pl-3 uppercase font-black text-md"
      >
        <a
          aria-current="page"
          className="bg-lime-500 text-slate-50 group flex items-center pl-5 py-3 hover:text-slate-900"
          href="/home"
        >
          Cars
        </a>
        <a
          className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
          href="reserve"
        >
          Reserve
        </a>
        <a
          className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
          href="/reservations"
        >
          My reservations
        </a>
        <a
          className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
          href="add"
        >
          Add car
        </a>
        <a
          className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
          href="delete"
        >
          Delete car
        </a>
        <span
          className="logout"
        >
          <button
            className="logout"
            onClick={[Function]}
            type="button"
          >
            <h4>
              LOGOUT
            </h4>
          </button>
        </span>
      </div>
    </nav>
  </div>
  <div
    className="flex-shrink-0 flex-200 p-4"
  >
    <div
      className="flex gap-2"
    >
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-6"
          src="twitter-icon.png"
        />
      </a>
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-6"
          src="facebook-icon.png"
        />
      </a>
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-6"
          src="linkedin-icon.png"
        />
      </a>
      <a
        href="/"
      >
        <img
          alt="social-icon"
          className="w-6"
          src="github-icon.png"
        />
      </a>
    </div>
    <p
      className="text-gray-700 font-bold mt-2"
    >
      Â© 2022
    </p>
  </div>
</div>
`;
